File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        int n = s.size();
        long long ans = 0;
        long long count = 0;
        for(int i = n-1; i >= 0; i--){
            if(s[i] == '0'){
                count++;
            }else{
                ans += count;
            }
        }
        return ans;
    }
};

File: 3195-separate-black-and-white-balls/separate-black-and-white-balls.py
class Solution:
    def minimumSteps(self, s: str) -> int:
        count = 0
        ans = 0
        for i in range(len(s) - 1,-1, -1):
            if s[i] == '0':
                count += 1
            else:
                ans += count
        return ans